server.js

```diff
@@ -16,8 +19,59 @@ const welcomeMessage = {
 //Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
+app.get("/", function (req, res) {
+  res.sendFile(__dirname + "/index.html");
+});
+
+// read all messages
+app.get("/messages", function(req,res) {
+  res.json(messages);
+})
+
+// read a message by id
+app.get("/messages/:id", function(req, res) {
+  const messageId = req.params.id;
+  const findId = messages.find((message) => { return message.id === Number(messageId) });
+    if (findId) {
+    res.json(findId);
+    console.log(findId);
+  } else {
+    res.status(404).send("Message not found");
+  }
+});
+
+
+// create a new message
+app.post("/messages", function (req, res) {
+  const {from, text} = req.body
+  const newMessage = {
+     id: messages.length,
```

Comment:
When setting the ID this way, what would happen if you add 3 messages, deleted the first one, then tried to add another?

ie messages = [1, 2, 3] then remove one -> [2, 3] then try to add a new one -> [2, 3, ?]